### PR TITLE
docs: add npm packages docs

### PR DESCRIPTION
**Pull request by Timeless0911**
## Summary

add npm packages docs

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/plugin-dts/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "rsbuild-plugin-dts",
   "version": "0.0.14",
-  "description": "Dts plugin for Rsbuild",
+  "description": "Rsbuild plugin that provides support for emitting declaration files for Typescript in Rslib",
```
Comment (fi3ework) [inline packages/plugin-dts/package.json:4]:
```suggestion
  "description": "Rsbuild plugin that supports emitting declaration files for TypeScript in Rslib.",
```

And, does it implicit it only works with Rslib?

File: website/docs/en/guide/start/npm-packages.mdx [hunk 2]
```diff
@@ -1 +1,30 @@
...
+Rslib core package that provides CLI commands and build capabilities based on Rsbuild.
+
+- [npm](https://npmjs.com/package/@rslib/core)
+- [Source Code](https://github.com/web-infra-dev/rslib/tree/main/packages/core)
+
+## rsbuild-plugin-dts
+
+![](https://img.shields.io/npm/v/rsbuild-plugin-dts?style=flat-square&colorA=564341&colorB=F8F5FF)
+
+Rsbuild plugin that provides support for emitting declaration files for Typescript in Rslib.
```
Comment (fi3ework) [inline website/docs/en/guide/start/npm-packages.mdx:18]: sync with above

File: website/docs/en/guide/start/npm-packages.mdx [hunk 3]
```diff
@@ -1 +1,30 @@
...
+
+- [npm](https://npmjs.com/package/rsbuild-plugin-dts)
+- [Source Code](https://github.com/web-infra-dev/rslib/tree/main/packages/plugin-dts)
+
+## create-rslib
+
+![](https://img.shields.io/npm/v/create-rslib?style=flat-square&colorA=564341&colorB=F8F5FF)
+
+Used to create a new Rslib project.
```
Comment (fi3ework) [inline website/docs/en/guide/start/npm-packages.mdx:27]: Scaffolding for Rslib projects.

Comment (Timeless0911) [reply website/docs/en/guide/start/npm-packages.mdx:27]: I just follow Rsbuild docs about this package.

File: packages/plugin-dts/package.json (hunk 1 above)
Comment (Timeless0911) [reply packages/plugin-dts/package.json:4]: It works in Rsbuild project, may be we can delete `in Rslib`.

**Review by fi3ework — APPROVED** [review]

**Pull request merged**